tile-service/src/main/java/com/oculusinfo/tile/rest/tile/TileServiceImpl.java

```diff
@@ -109,6 +110,60 @@ public TileServiceImpl( LayerService layerService ) {
 		return data;
 	}
 
+	/**
+	 * TEMPORARY - checks the pyramid IO and modifies the tile name to include the slice range if SliceHBasePyramidIO
+	 * is used.  This needs to be replaced with a more generic request pre-process method can instantiated through
+	 * config.
+	 */
+	private <T> List<TileData<T>> getTileDatas(String layer, String dataId, TileSerializer<T> serializer, PyramidIO pyramidIO,
```

Comment:
I'm concerned about having both 'layer' and 'dataId' here.

Reading the code, I think 'layer' is being misused, and all uses of it should be using 'dataId' instead.

Not particularly: 'dataId' is being used for the readTiles call, but 'layer' is being used for the getMetaData call - those two calls should use the same ID.

I haven't checked every use - just what I've stated above - but it's enough to make me suspicious of them all.